### PR TITLE
About info in Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As a user I should be able to check the about section so I can learn more about Target MVD](https://trello.com/c/bZ4Or9aW)

---

#### Description:

* Add endpoint to return general information about the site base on a key
* Add request spec test to validate the general information endpoint
* Add default value to general information table

### CODE REVIEW

File: app/controllers/api/v1/general_informations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Api
+  module V1
+    class GeneralInformationsController < ApiController
+      def show
+        @info = GeneralInformation.find_by! key: params[:id]
```
Comment (gabrielbursztein2) [inline app/controllers/api/v1/general_informations_controller.rb]: It should be `GeneralInformation.find_by(key: params[:key])` right?

File: app/views/api/v1/general_informations/_info.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,2 @@
+json.title    info.title
```
Comment (gabrielbursztein2) [inline app/views/api/v1/general_informations/_info.json.jbuilder]: There is no need to have a partial if you plan on using it only once.

File: app/views/api/v1/general_informations/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1 @@
+json.generalInformation @info, partial: 'info', as: :info
```
Comment (gabrielbursztein2) [inline app/views/api/v1/general_informations/show.json.jbuilder]: Always use snake_case

**Review by gabrielbursztein2 — COMMENTED** [review]
Nice!

File: app/views/api/v1/general_informations/show.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,4 @@
+json.general_information do
+  json.title    @info.title
```
Comment (JuanfraM) [inline app/views/api/v1/general_informations/show.json.jbuilder:2]: It'd be better if this view calls a partial `_info` so this partial can be reused

File: config/routes.rb [hunk 5]
```diff
@@ -8,6 +8,7 @@
       devise_scope :user do
         resources :users, only: %i[update]
         resources :targets, only: %i[index create destroy]
+        resources :general_informations, only: %i[show], param: :key
```
Comment (JuanfraM) [inline config/routes.rb]: Just to let you know when u have just one element you can omit the array and use `only: :show`

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+GeneralInformation.create!(
```
Comment (JuanfraM) [inline db/seeds.rb]: Use `finde_or_create_by` in case you want to run the seed again with other data

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,8 @@
...
+  key: 'about',
+  title: 'WHAT’S TARGET?',
```
Comment (JuanfraM) [inline db/seeds.rb]: double quotes?

File: spec/request/api/v1/general_informations/show_spec.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      end
+
+      it 'is expected that response contains some body data' do
+        body = (JSON response.body)
+        expect(json_value(body, 'general_information', 'title')).not_to be_nil
+        expect(json_value(body, 'general_information', 'text')).not_to be_nil
+      end
+    end
+
+    context 'when the key doesn\'t exist' do
```
Comment (JuanfraM) [inline spec/request/api/v1/general_informations/show_spec.rb]: it's clearer => `context "when the key doesn't exist" do`

**Review by JuanfraM — APPROVED** [review]

**Review by gabrielbursztein2 — APPROVED** [review]

**Pull request merged**